src/Mod/PartDesign/Gui/TaskHoleParameters.cpp

```diff
@@ -932,21 +957,14 @@ void TaskHoleParameters::changedObject(const App::Document&, const App::Property
         ui->Depth->setDisabled(ro);
     }
     else if (&Prop == &hole->DrillPoint) {
-        ui->drillPointFlat->setEnabled(true);
-        ui->drillPointAngled->setEnabled(true);
         std::string drillPoint(hole->DrillPoint.getValueAsString());
-        if (drillPoint == "Flat" && !ui->drillPointFlat->isChecked()) {
-            ui->drillPointFlat->blockSignals(true);
-            ui->drillPointFlat->setChecked(true);
-            ui->drillPointFlat->blockSignals(false);
-        }
-        if (drillPoint == "Angled" && !ui->drillPointAngled->isChecked()) {
-            ui->drillPointAngled->blockSignals(true);
-            ui->drillPointAngled->setChecked(true);
-            ui->drillPointAngled->blockSignals(false);
+        ui->DrillPointAngled->setEnabled(true);
+        if (ui->DrillPointAngled->isChecked() ^ (drillPoint == "Angled")) {
+            ui->DrillPointAngled->blockSignals(true);
+            ui->DrillPointAngled->setChecked(drillPoint == "Angled");

```

Comment:
It probably doesn't make any difference (compilers are pretty smart), but I'd probably change the definition from a string to a boolean, and only do the comparison once. Then directly use the boolean value in the two lines below.

Reply:
since currently there are only 2 options it makes sense, but i think this was this done to have multiple type of end cuts, like the comment [here](https://github.com/FreeCAD/FreeCAD/pull/19167#issuecomment-2613929135) mentions, i've seen that on other cad packages